app/src/main/java/me/quadphase/qpdex/AdvancedSearch.java

```diff
@@ -0,0 +1,508 @@
+        eggGroup1Spinner.setAdapter(eggGroupArrayAdapter);
+        eggGroup2Spinner.setAdapter(eggGroupArrayAdapter);
+        generationSpinner.setAdapter(generationArrayAdapter);
+
+        type1Spinner.setOnItemSelectedListener(new DropDownSelectSpinner());
+        type2Spinner.setOnItemSelectedListener(new DropDownSelectSpinner());
+        abilitySpinner.setOnItemSelectedListener(new DropDownSelectSpinner());
+        eggGroup1Spinner.setOnItemSelectedListener(new DropDownSelectSpinner());
+        eggGroup2Spinner.setOnItemSelectedListener(new DropDownSelectSpinner());
+        generationSpinner.setOnItemSelectedListener(new DropDownSelectSpinner());
+
+        long endTime = System.nanoTime();
+        long duration = (endTime - startTime);
+
+        Log.d("QPDex", String.format("Total filling spinners took %s ns",duration));
+    }
+
+    private void resetFilter(){
+        // Provide a set of solutions if no options are selected
+        filteredNationalIds = new ArrayList<Integer>();
+        filteredUniqueIds = new ArrayList<Integer>();
+        currentFilterPokemonArrayList = new ArrayList<Integer>();
+
+        for(int i =1; i< pokemonFactory.getMAX_UNIQUE_ID()+1 ; i++ ){
+            filteredUniqueIds.add(i);
+            currentFilterPokemonArrayList.add(i);
+        }
+
+        for (int i=1; i <pokemonFactory.getMAX_NATIONAL_ID()+1; i++){
+            filteredNationalIds.add(i);
+        }
+    }
+    @Override
+    public boolean onCreateOptionsMenu(Menu menu) {
+        // Inflate the menu; this adds items to the action bar if it is present.
+        getMenuInflater().inflate(R.menu.menu_advanced_search, menu);
+        return true;
+    }
+
+    @Override
+    public boolean onOptionsItemSelected(MenuItem item) {
+        // Handle action bar item clicks here. The action bar will
+        // automatically handle clicks on the Home/Up button, so long
+        // as you specify a parent activity in AndroidManifest.xml.
+        int id = item.getItemId();
+
+        //noinspection SimplifiableIfStatement
+        if (id == R.id.action_settings) {
+            return true;
+        }
+
+        return super.onOptionsItemSelected(item);
+    }
+
+    private class DropDownSelectSpinner implements AdapterView.OnItemSelectedListener {
+
+        /**
+         * Saves the chosen selections from the dropdown spinners
+         *
+         * @param parent either the type1 or type spinner
+         * @param view
+         * @param pos corresponds to the typeID position
+         * @param id corresponds to the id of the position
+         */
+        public void onItemSelected(AdapterView<?> parent, View view,
+                                   int pos, long id) {
+            // An item was selected. You can retrieve the selected item using
+            // parent.getItemAtPosition(pos)
+
+            // Determine other type, and set current type
+            Type otherType = new Type("Temp", -1);

```

Comment:
I'm not clear on how otherType is used. It is a local variable to this method that is never read.
Is it meant to make it so that you can't select the same type twice?